infrastructure/create-github-environment.js

```diff
@@ -0,0 +1,226 @@
+const { writeFileSync } = require('fs')
+
+const config = {
+  environment: '',
+  repo: {
+    REPOSITORY_ID: '',
+    DOCKERHUB_ACCOUNT: '', // This may be a dockerhub organisation or the same as the username
+    DOCKERHUB_REPO: '',
+    DOCKER_USERNAME: process.env.DOCKER_USERNAME,
+    DOCKER_TOKEN: process.env.DOCKER_TOKEN
+  },
+  ssh: {
+    KNOWN_HOSTS: process.env.KNOWN_HOSTS,
+    SSH_HOST: process.env.SSH_HOST, // IP address for the manager
+    SSH_USER: process.env.SSH_USER,
+    // SUDO_PASSWORD: process.env.SUDO_PASSWORD, // in case your user is not root
+    SSH_KEY: process.env.SSH_KEY // id_rsa
+  },
+  infrastructure: {
+    DISK_SPACE: '',
+    HOSTNAME: '', // server machine hostname used when provisioning - TODO: Adapt to support 3 or 5 replicas
+    DOMAIN: '', // web hostname applied after all public subdomains in Traefik,
+    REPLICAS: '1' // TODO: Adapt to support 3 or 5 replicas
+  },
+  services: {
+    SENTRY_DSN: process.env.SENTRY_DSN || '',
+    ELASTALERT_SLACK_WEBHOOK: process.env.ELASTALERT_SLACK_WEBHOOK || '',
+    INFOBIP_API_KEY: process.env.INFOBIP_API_KEY || '',
+    INFOBIP_GATEWAY_ENDPOINT: process.env.INFOBIP_GATEWAY_ENDPOINT || '',
+    INFOBIP_SENDER_ID: process.env.INFOBIP_SENDER_ID || '' // the name of the SMS sender e.g. OpenCRVS
+  },
+  seeding: {
+    ACTIVATE_USERS: true,
+    AUTH_HOST: '',
+    COUNTRY_CONFIG_HOST: '',
+    GATEWAY_HOST: ''
+  },
+  smtp: {
+    SMTP_HOST: process.env.SMTP_HOST || '',
+    SMTP_USERNAME: process.env.SMTP_USERNAME || '',
+    SMTP_PASSWORD: process.env.SMTP_PASSWORD || '',
+    EMAIL_API_KEY: process.env.EMAIL_API_KEY || '',
+    SMTP_PORT: '',
+    ALERT_EMAIL: ''
+  },
+  vpn: {
+    // openconnect details for optional VPN
+    VPN_PROTOCOL: '', // e,g, fortinet, wireguard etc
+    VPN_HOST: process.env.VPN_HOST || '',
+    VPN_PORT: process.env.VPN_PORT || '',
+    VPN_USER: process.env.VPN_USER || '',
+    VPN_PWD: process.env.VPN_PWD || '',
+    VPN_SERVERCERT: process.env.VPN_SERVERCERT || ''
+  },
+  whitelist: {
+    CONTENT_SECURITY_POLICY_WILDCARD: '', // e.g. *.<your-domain>
+    CLIENT_APP_URL: '',
+    LOGIN_URL: ''
+  },
+  backup: {
+    BACKUP_HOST: process.env.BACKUP_HOST || '',
+    BACKUP_DIRECTORY: '',
+    qa: {
+      RESTORE_DIRECTORY: '' // If making use of script to restore a production backup on QA for regular monitoring
+    }
+  }
+}
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN
+})
+
+async function createVariable(environment, name, value) {
+  await octokit.request(
+    `POST /repositories/${config.repo.REPOSITORY_ID}/environments/${config.environment}/variables`,
+    {
+      repository_id: config.repo.REPOSITORY_ID,
+      environment_name: environment,
+      name: name,
+      value: value,
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    }
+  )
+}
+
+async function createSecret(environment, key, keyId, name, secret) {
+  //Check if libsodium is ready and then proceed.
```

Comment:
Error: Cannot find module 'libsodium-wrappers'
Require stack:
- /Users/naf/Projects/opencrvs-cdpi-living-lab/infrastructure/create-github-environment.js

---

think we need to amend package.json